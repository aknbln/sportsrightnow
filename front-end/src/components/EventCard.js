import React from "react";

import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import { Navigate, useNavigate, useSearchParams } from "react-router-dom";
import { Highlight } from "../Utils";

const EventCard = (props) => {
  const { name, local_date, city, local_time, venue, id } = props.eventData;
  const search = [props.q]
  const navigate = useNavigate()


  function ButtonClicked(id) {
    navigate("/events/instance?id=" + id)
  }

  return (
    <Card
    style= {{backgroundColor: 'lightgray', width: "120%"}}>
      <Card.Body>
        <Card.Header style = {{fontSize: '3vh', color: '#333232'}}>{Highlight(name, search)}</Card.Header>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>Date: {Highlight(local_date, search)}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>Time: {Highlight(local_time, search)}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>City: {Highlight(city, search)}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>Venue: {Highlight(venue, search)}</Card.Text>
        
      </Card.Body>
      <Card.Footer style= {{backgroundColor: 'dimgray'}}>
        <Button class="btn btn-primary stretched-link" onClick={() => ButtonClicked(id)}
        style= {{backgroundColor: '#3d405b', borderColor: 'black'}}>
          More Info
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default EventCard;