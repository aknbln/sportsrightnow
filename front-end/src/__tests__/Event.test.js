// Replace useNavigates with a dummy function as it doesn't work without a <Router>

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

import renderer from 'react-test-renderer'
import EventCard from '../components/EventCard'

const Event = 
{
    name: "Buffalo Ballers vs Jovian Jabbers",
    date: "1/01/1984",
    location: "Jupiter",
    venue: "Europa Gigadome",
    league: "NBA",
    event_id: "999",
}

it('EventCard data correct', () => {
    const component = renderer.create(<EventCard eventData={Event}/>)
    expect(component.location = "Jupiter")
})

it('EventCard renders correctly', () => {
    const component = renderer.create(<EventCard eventData={Event}/>)
    let json = component.toJSON()
    expect(json).toMatchSnapshot()
})