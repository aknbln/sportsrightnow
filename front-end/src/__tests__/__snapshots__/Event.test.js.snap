// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "backgroundColor": "lightgray",
    }
  }
>
  <div
    className="card-body"
  >
    <div
      className="card-header"
      style={
        Object {
          "color": "#333232",
          "fontSize": "3vh",
        }
      }
    >
      Buffalo Ballers vs Jovian Jabbers
    </div>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      Date: 
      1/01/1984
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      City: 
      Jupiter
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      Venue: 
      Europa Gigadome
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      League: 
      NBA
    </p>
  </div>
  <div
    className="card-footer"
    style={
      Object {
        "backgroundColor": "dimgray",
      }
    }
  >
    <a
      class="btn btn-primary stretched-link"
      className="btn btn-primary"
      disabled={false}
      href="/events/999"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "backgroundColor": "#3d405b",
          "borderColor": "black",
        }
      }
      tabIndex={0}
    >
      More Info
    </a>
  </div>
</div>
`;
